### ISSUE TITLE
Bug with Experimental Feature: BeginComboPreview

### ISSUE THREAD

**Issue by BobbyAnguelov** [ocornut/imgui#6501]
_(you may also go to Demo>About Window, and click "Config/Build Information" to obtain a bunch of detailed information that you can paste here)_

**Version/Branch of Dear ImGui:**

```
Dear ImGui 1.89.6 (18960)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: _WIN32
define: _WIN64
define: _MSC_VER=1936
define: _MSVC_LANG=201402
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_win32
io.BackendRendererName: imgui_impl_dx11
io.ConfigFlags: 0x00000443
 NavEnableKeyboard
 NavEnableGamepad
 DockingEnable
 ViewportsEnable
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00001C0E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 HasMouseHoveredViewport
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1264.00,761.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```
Operating System: Windows11

**My Issue/Question:**

So this is super weird, but using the code below:

```c++
ImGui::Begin( "Repro" );
if ( ImGui::BeginChild( "DAS", ImVec2( -1, ImGui::GetFrameHeightWithSpacing() ) ) )
{
    ImGui::SetNextItemWidth( -1 );
    if ( ImGui::BeginCombo( "##Test", "", ImGuiComboFlags_CustomPreview ) )
    {
        ImGui::EndCombo();
    }

    if ( ImGui::BeginComboPreview() )
    {
        ImGui::Text( "Preview" );
        ImGui::EndComboPreview();
    }
}
ImGui::EndChild();
ImGui::End();
```

This works when the imgui window is on my main monitor but if I move it to a secondary monitor (to the left so negative x coords), then preview stops displaying due to the narrow test on line 1838 in imgui_widgets.cpp failing. Shouldn't that line read:

```c++
if (!window->ClipRect.Overlaps(preview_data->PreviewRect))
```
instead of 
```c++
if (!window->ClipRect.Contains(preview_data->PreviewRect))
```

Changing that fixes the issue. Also this only seems to occur when you put the combo inside of a child window.

**Comment by ocornut** [comment]
You are correct. I have pushed your fix as a134892
(It's not merged in docking until a few days but I have confirmed it cherry-pick fines on docking, if you need a merge sooner let me know)

Thank you for reporting and fixing!

**Comment by ocornut** [comment]
Cross-posting for visibility:
Info: `BeginComboPreview()` has been updated in 1.93.0 WIP (IMGUI_VERSION_NUM >= 19296)
The `ImGuiComboFlags_CustomPreview` flag is not needed anymore. Read details at:
https://github.com/ocornut/imgui/issues/1658#issuecomment-5589399024
